### PR TITLE
chore(packages): add func to expose CW operation outcomes to doc queries

### PR DESCRIPTION
**Pull request by Goncharo**
refs. metriport/metriport-internal#585

### Dependencies

- Downstream: TBD

### Description

add func to expose CW operation outcomes to doc queries - updating API to use this will be done in another issue

### Release Plan

- [ ] merge this PR
- [ ] merge downstream

### CODE REVIEW

**Review by Orta21 — APPROVED** [review]

File: packages/packages/commonwell-sdk/src/client/commonwell.ts [hunk 1]
```diff
@@ -768,6 +768,23 @@ export class CommonWell implements CommonWellAPI {
...
+   * @see {@link https://specification.commonwellalliance.org/services/data-broker/cha-broker-api-reference#104-document-query|Use case}
+   * @see {@link https://specification.commonwellalliance.org/services/data-broker/protocol-operations-data-broker#8781-find-documents|API spec}
+   */
+  async queryDocumentsFull(
+    meta: RequestMetadata,
+    patientId: string
+  ): Promise<DocumentQueryFullResponse> {
+    const headers = await this.buildQueryHeaders(meta);
+    return document.queryFull(this.api, headers, patientId);
+  }
```
Comment (leite08) [inline packages/packages/commonwell-sdk/src/client/commonwell.ts:786]: What do you think we add an optional `options` param to `queryDocuments` to indicate whether it should return only documents (filtering out invalid ones) or include errors/operation outcomes as well, something like:

```
queryDocuments(meta: RequestMetadata, patientId: string, options?: { includeFailures: boolean })
```

It's easier from the SDK user's perspective to have one function to return docs and just parameterize it accordingly. I understand we'd have a diff return type on `queryDocuments`, but I think its worth it.

**Review by leite08 — COMMENTED** [review]
Suggesting we update the existing function 👍

File: packages/packages/commonwell-sdk/src/client/commonwell.ts (hunk 1 above)
Comment (Goncharo) [reply packages/packages/commonwell-sdk/src/client/commonwell.ts:786]: To clarify the intention here - in a subsequent PR I'll remove the one that only gets doc refs, as I don't see a good use case for it. Just wanted to get this out the door quicker before also doing a bit of a refactoring on the API.

File: packages/packages/commonwell-sdk/src/client/document.ts [hunk 2]
```diff
@@ -33,6 +38,33 @@ export async function query(
...
+export async function queryFull(
+  api: AxiosInstance,
+  headers: Record<string, string>,
+  patientId: string
+): Promise<DocumentQueryFullResponse> {
+  const subjectId = convertPatientIdToSubjectId(patientId);
+  if (!subjectId) {
+    throw new Error(`Could not determine subject ID for document query`);
+  }
+  const url = `${CommonWell.DOCUMENT_QUERY_ENDPOINT}?subject.id=${subjectId}`;
```
[the remark this answers is not in the document]
Comment (Goncharo) [reply packages/packages/commonwell-sdk/src/client/document.ts:50]: See comment [here](https://github.com/metriport/metriport/pull/246#discussion_r1174427060) - I will be removing the old function when I do the API refactoring

File: packages/packages/commonwell-sdk/src/client/commonwell.ts (hunk 1 above)
Comment (leite08) [reply packages/packages/commonwell-sdk/src/client/commonwell.ts:786]: Sounds good.

**Review by leite08 — APPROVED** [review]

**Pull request merged**